services/galley/src/Galley/API/Teams/Features.hs

```diff
@@ -124,23 +124,29 @@ setFeatureStatus ::
       '[ Error ActionError,
          Error TeamError,
          Error NotATeamMember,
-         TeamStore
+         TeamStore,
+         Error TeamFeatureError
        ]
       r
   ) =>
+  (GetFeatureInternalParam -> Galley r (Public.TeamFeatureStatus a)) ->
   (TeamId -> Public.TeamFeatureStatus a -> Galley r (Public.TeamFeatureStatus a)) ->
   DoAuth ->
   TeamId ->
   Public.TeamFeatureStatus a ->
   Galley r (Public.TeamFeatureStatus a)
-setFeatureStatus setter doauth tid status = do
+setFeatureStatus getter setter doauth tid status = do
```

Comment:
I don't think we have to (or want to) change the signature of `iPutTeamFeature`.  Instead of `getter`, I would write a `get`-variant of `Galley.Cassandra.TeamFeatures.setPaymentStatus` and call that directly.  All you need for that is `a` from the type signature here.

This would make a lot of noise in this change evaporate.

Reply:
I wasn't sure, but yes this is an option, and I'll go for that then. Thanks!